javascript/code-challenges/stack-queue-animal-shelter/README.md

```diff
@@ -0,0 +1,57 @@
+# Animal Shelter
+
+## Implements an Animal Shelter Class
+
+This class stores objects that denote a cat or a dog and maintains a FIFO order
+
+## Challenge
+
+Create a class called AnimalShelter which holds only dogs and cats.
+The shelter operates using a first-in, first-out approach.
+Implement the following methods:
+
+enqueue, dequeue
+
+Tests were written to prove the following functionality:
+
+    Shelter allows the dequeuing of one animal type mixed in with others without changing the FIFO order
+    Shelter allows the enqueuing and dequeuing of either a dog or cat
+    The dequeue method returns null if the user enters an animal that is not a cat or dog when dequeueing
+    Shelter does not allow the dequeuing of animals if there are no animals in it
+    Dequeue method returns null if a valid animal type if preferred, but no animals of that type are in the shelter
+
+## Approach & Efficiency
+<!-- What approach did you take? Why? What is the Big O space/time for this approach? -->
+My approach was to find what combination of stacks and queues would work best in terms of efficiency. I ended up implementing this solution with 2 queues. The challenge here was maintaining the FIFO order when the first item wasn't the correct animal. I ended up moving all the data to a 2nd queue when this happened to maintain order integrity which was still relatively efficient.
+
```

Comment:
I think two queues would be the best approach however it might have been easier to have one queue for dogs and one for cats. Then when there's an adoption request for a dog, you could just dequeue from that queue. But overall, nice work!

Reply:
That's a much better approach, thanks for the tip!